src/Azure.Deployments.Extensibility.AspNetCore/Middlewares/ErrorResponseExceptionHandler.cs

```diff
@@ -0,0 +1,31 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Azure.Deployments.Extensibility.AspNetCore.Exceptions;
+using Microsoft.AspNetCore.Diagnostics;
+using Microsoft.AspNetCore.Http;
+using System.Net;
+
+namespace Azure.Deployments.Extensibility.AspNetCore.Middlewares
+{
+    public class ErrorResponseExceptionHandler : IExceptionHandler
+    {
+        public async ValueTask<bool> TryHandleAsync(HttpContext httpContext, Exception exception, CancellationToken cancellationToken)
+        {
+            if (exception is ErrorResponseException errorResponseException)
+            {
+                await this.HandleErrorResponseException(httpContext, errorResponseException);
+
+                return true;
+            }
+
+            return false;
+        }
+
+        protected virtual async Task HandleErrorResponseException(HttpContext httpContext, ErrorResponseException errorResponseException)
+        {
+            httpContext.Response.StatusCode = (int)HttpStatusCode.BadRequest;
```

Comment:
>BadRequest [](http://example.com/codeflow?start=66&length=10)

So all `ErrorResponseException` exceptions return 400s?

Reply:
Yeah, but the extension author may choose to override it.